### ISSUE TITLE
Can't run the MacOS DMG

### ISSUE THREAD

**Issue by j3pic** [shikantaza/pLisp#54]
MacOS reports that it "quit unexpectedly". I tried running it from a terminal, and it gave me this:

    2019-07-07 01:42:52.423 defaults[44952:16389959]
    The domain/default pair of (.GlobalPreferences, AppleCollationOrder) does not exist
    -----------------------------INFO------------------------------
    Default image file 'plisp.image' not found in the current
    directory. Working with an image is faster and more convenient,
    as the core library doesn't have to be recompiled each time
    plisp is invoked. You can create the image file by choosing
    the 'Save image' option in the Transcript window, and invoking
    plisp as 'plisp -i <image file>' thereafter. If you name your
    image file as 'plisp.image', you do not have to specify it
    during startup; pLisp will use this file automatically.
    ---------------------------------------------------------------
    
    2019-07-07 01:42:52.773 pLisp-bin[44947:16390001] *** WARNING: Method userSpaceScaleFactor in class NSView is deprecated on 10.7 and later. It should not be used in new applications. Use convertRectToBacking: instead.
    Loading core library (this may take a while)... done
    
    (pLisp-bin:44947): Gtk-WARNING **: Error loading icon from file '/Applications/pLisp.app/Contents/MacOS/../Resources/share/plisp/icons/evaluate.png':
    	Unable to load image-loading module: Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so: dlopen(Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so, 1): image not found
    
    (pLisp-bin:44947): Gtk-WARNING **: Could not load a pixbuf from icon theme.
    This may indicate that pixbuf loaders or the mime database could not be found.
    **
    Gtk:ERROR:gtkiconhelper.c:493:ensure_surface_for_gicon: assertion failed: (destination)
    Abort trap: 6

libpixbufloader-png.so is installed at /Applications/pLisp.app/Contents/Resources/lib/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-png.so, so I don't know why it couldn't open that library.

**Comment by shikantaza** [comment]
Can you run pLisp from the terminal with the -c option and see if the
console mode works? I want to check if the abort trap is because of the
libpixbufloader problem or something else.

Thanks,
Rajesh

**Comment by j3pic** [comment]
With the -c option I was able to get to a `USER>` prompt and evaluate a couple of simple expressions.

**Comment by j3pic** [comment]
I discovered that MAKE-SYMBOL is not a function. So I tried to evaluate `make-symbol` and pLisp just froze.

**Comment by j3pic** [comment]
In fact, in pLisp -c, evaluating anything that's an undefined variable freezes the interpreter.

**Comment by shikantaza** [comment]
Thanks for the update, will investigate this and get back to you.

Regards,
Rajesh

**Comment by shikantaza** [comment]
Apologies, I have not been able to make much headway in getting to the
bottom of this. It could be because of a version mismatch in the libpng
library, per my google-fu. Unfortunately I do not have a clean OS X
installation (without the contamination of the dev/compile environment), so
the fact that I am able to run pLisp successfully from the DMG installation
does not preclude the absence of fundamental issues with the OS X build.
I'll anyway shortly create a new DMG based on the latest commit and let you
know (and also keep my fingers crossed) .

Regards,
Rajesh

**Comment by shikantaza** [comment]
I have pushed a new DMG file (v0.1.26), can you please check if the issue
occurs in this version too? Also, if you could let me know your OS X
version, I'll see if I can get a VirtualBox image and try to replicate the
issue.

Thanks,
Rajesh

**Comment by shikantaza** [comment]
Yup, there is an issue. I'm get the same error on a virgin High Sierra VM.
Will work on this and update you. Thanks for your patience.

Regards,
Rajesh

**Comment by shikantaza** [comment]
Turns out the OS X DMG creation process (using gtk-mac-builder) had issues with hardcoding of paths from the development environment for gdk-pixbuf. Also, gtksourceview styles and schema files were missing from the bundle. Fixed these issues and was able to successfully run pLisp in a clean High Sierra VM. The latest OS X DMG incorporates these fixes, can you please check? I'll close the issue once I get the confirmation that things work as advertised now.

Regards,
Rajesh
